### PR TITLE
Add Json property type with dot-path query support (unified-refs-edges-json phase 1)

### PR DESCRIPTION
**Pull request by dnorman**
Introduces `ankurah::property::Json` for storing structured data with AnkQL path queries (e.g., `context.session_id = 'xyz'`). This implements Phase 1a (parser paths) and Phase 1b (Json type) of the unified-refs-edges-json spec.

### AnkQL Parser

- **PathExpr AST**: Replaces `Identifier` enum with `PathExpr { steps: Vec<String> }` for unified path handling
- **Dot-path syntax**: Parse multi-step navigation like `licensing.territory` or `context.user_id`
- **Grammar**: `PathExpr = { Identifier ~ ("." ~ Identifier)* }`

### Json Property Type

- **`ankurah::property::Json`**: First-class property type wrapping `serde_json::Value`
- **LWW backend**: Last-writer-wins semantics by default
- **Path extraction**: `get_path()` and `value_at_path()` for nested structure access
- **Conversions**: `From<serde_json::Value>` / `Into<serde_json::Value>`

### Filter Evaluation

- **JSON traversal**: Multi-step path navigation through JSON structures
- **Type-aware comparison**: Handles JSON numbers, strings, booleans, nulls with appropriate coercion
- **Consistent behavior**: Same semantics across all backends

### Backend Support

- **Postgres**: JSONB column type with `->>` and `->` operators for path queries
- **Sled**: JSON path extraction in filter evaluation  
- **IndexedDB**: Full path traversal with type coercion for browser-side queries

### WASM / TypeScript

- **Direct marshaling**: `Json` converts to/from native JS objects via `serde-wasm-bindgen`
- **TypeScript type**: `export type Json = any;`
- **Generated bindings**: `LWWJson.get()` returns JS object, `LWWJson.set()` accepts JS object

### Value Ergonomics

- `Value::json<T: Serialize>()` — serialize any type to `Value::Json`
- `Value::parse_as_json<T: DeserializeOwned>()` — deserialize from Json/Binary/String variants
- `Value::parse_as_string<T: FromStr>()` — parse string-based enums
- `impl From<serde_json::Error> for PropertyError` — ergonomic error handling

### Other Improvements

- **EntityId**: `FromStr` impl and `From<EntityId> for String` for easier conversions

### Tests

- `pg_json_property.rs` — Postgres JSONB queries with path navigation
- `pg_jsonb_semantics.rs` — Type coercion and edge cases
- `sled_json_property.rs` — Sled filter evaluation
- `indexeddb json_property.rs` — WASM browser-side tests
- Selection macro tests for path syntax

### Breaking Changes

None. All changes are additive.

### Related

- Spec: `specs/unified-refs-edges-json/phase-1a-parser-paths.md`
- Spec: `specs/unified-refs-edges-json/phase-1b-json-type.md`
- Spec: `specs/unified-refs-edges-json/phase-1c-structured-queries.md`

### CODE REVIEW

File: specs/unified-refs-edges-json/phase-1a-parser-paths.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+## Phase 1a: Parser — Basic Dot-Path Navigation
+
+### Goal
+
+Extend AnkQL to parse multi-step dot paths like `licensing.territory`. This is the minimal parser change needed for structured property queries.
```
Comment (mmacfadden) [inline specs/unified-refs-edges-json/phase-1a-parser-paths.md]: What about array indices?
